### ISSUE TITLE
为什么当前的线性分类器是里

### ISSUE THREAD

**Issue by stewnn-alt** [baichuan-inc/Baichuan2#189]
Since the current linear classifier computes logits by dot product, which is a mixture of L2 distance and cosine similarity.

**Comment by mmmans** [comment]
dot product :  d( q , k) = || q || * || k || * cosine(q,k)
l2 dist(不开方） l(q,k) = || q - k ||^2 = ||q||^2 + ||k||^2 - 2cosine(q,k)
cosine similarity  sim(q,k) = cosine(q,k)

**Comment by stewnn-alt** [comment]
了解了，谢谢~
